src/main/java/com/crowdin/cli/commands/actions/DownloadReviewedSourcesAction.java

```diff
@@ -0,0 +1,256 @@
+        this.noProgress = noProgress;
+        this.plainView = plainView;
+        this.branchName = branchName;
+    }
+
+
+    @Override
+    public void act(Outputter out, PropertiesWithFiles properties, ProjectClient client) {
+        this.out = out;
+        boolean isOrganization = PropertiesBeanUtils.isOrganization(properties.getBaseUrl());
+        if (!isOrganization) {
+            out.println(WARNING.withIcon(RESOURCE_BUNDLE.getString("error.only_enterprise")));
+            return;
+        }
+        if (!plainView && !properties.getPreserveHierarchy()) {
+            out.println(WARNING.withIcon(RESOURCE_BUNDLE.getString("message.download_sources.preserve_hierarchy_warning")));
+        }
+
+        CrowdinProjectFull project = ConsoleSpinner
+                .execute(out, "message.spinner.fetching_project_info", "error.collect_project_info",
+                        this.noProgress, this.plainView, client::downloadFullProject);
+
+        Long branchId = Optional.ofNullable(this.branchName)
+                .map(br -> project.findBranchByName(br)
+                        .orElseThrow(() -> new RuntimeException(RESOURCE_BUNDLE.getString("error.not_found_branch"))))
+                .map(Branch::getId)
+                .orElse(null);
+
+        BuildReviewedSourceFilesRequest request = new BuildReviewedSourceFilesRequest();
+        //branch id does not work properly
+//        request.setBranchId(branchId);
+        ReviewedStringsBuild build = this.buildReviewedSources(client, request);
+
+        String randomHash = RandomStringUtils.random(11, false, true);
+
+        String baseTemp = StringUtils.removeEnd(properties.getBasePath(), Utils.PATH_SEPARATOR) +
+                Utils.PATH_SEPARATOR +
+                "CrowdinReviewedSources_" +
+                randomHash;
+        File baseTempDir = new File(baseTemp + Utils.PATH_SEPARATOR);
+        String downloadedZipArchivePath = baseTemp + ".zip";
+        File downloadedZipArchive = new File(downloadedZipArchivePath);
+
+        this.downloadReviewedSources(client, build.getId(), downloadedZipArchivePath);
+
+        List<File> downloadedFiles = this.extractArchive(downloadedZipArchive, baseTempDir);
+
+        Map<String, File> downloadedReviewedFilesMap = downloadedFiles.stream()
+                .collect(Collectors.toMap(
+                        file -> {
+                            String path = StringUtils.removeStart(
+                                    file.getAbsolutePath(),
+                                    baseTempDir.getAbsolutePath() +
+                                            Utils.PATH_SEPARATOR +
+                                            project.getSourceLanguageId() + "-REV"
+                            );
+                            return Utils.unixPath(path);
+                        },
+                        Function.identity()
+                ));
+
+        //code below is a copy/paste from DownloadSourcesAction
```

Comment:
Any chance to extract somewhere this part of the code to avoid duplication?

Reply:
Or maybe make DownloadSourcesAction able to work with the new parameter (action will work just with the different zip archives depending on the presence of the `--reviewed` parameter)?